### PR TITLE
HYC - 1887 - Updated Analytics for Matomo Integration  

### PR DESCRIPTION
**Pull request by davidcam-src**
https://unclibrary.atlassian.net/browse/HYC-1887

* Download analytics sends a request to matomo for tracking instead of GA4
* Modified tests to reflect changes to download analytics
* Rolled back stats controller override for returning 404 on stats related pages
* Matomo override introduced for retrieval of page view and download stats
* Tracking tag with some changes to include custom dimensions included in `_matomo.html.erb`

### CODE REVIEW

File: app/controllers/concerns/hyc/download_analytics_behavior.rb [hunk 1]
```diff
@@ -45,15 +53,32 @@ def api_secret
         @api_secret ||= ENV['ANALYTICS_API_SECRET']
       end
 
+      def user_id
```
Comment (bbpennel) [inline app/controllers/concerns/hyc/download_analytics_behavior.rb:56]: don't think this is used anymore

File: app/views/hyrax/file_sets/_show_actions.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <%# [hyc-override] Overriding to make only admins able to delete a fileset %>
 <%# https://github.com/samvera/hyrax/tree/hyrax-v4.0.0/app/views/hyrax/file_sets/_show_actions.html.erb %>
 <div class="form-actions">
-  <% if Hyrax.config.analytics? && Hyrax.config.analytics_provider != 'ga4' %>
+  <% if Hyrax.config.analytics? %>
```
Comment (bbpennel) [inline app/views/hyrax/file_sets/_show_actions.html.erb:4]: same here

File: spec/fixtures/files/matomo_work_downloads.json
Comment (bbpennel) [inline spec/fixtures/files/matomo_work_downloads.json:1]: is this file still needed?

File: spec/fixtures/files/matomo_work_views.json
Comment (bbpennel) [inline spec/fixtures/files/matomo_work_views.json:1]: is this file still needed?

**Review by bbpennel — CHANGES_REQUESTED** [review]

File: app/views/hyrax/base/_show_actions.html.erb [hunk 3]
```diff
@@ -31,7 +31,7 @@
             class: presenter.display_unfeature_link? ? 'btn btn-secondary' : 'btn btn-secondary collapse' %>
       <% end %>
     <% end %>
-    <% if Hyrax.config.analytics? && Hyrax.config.analytics_provider != 'ga4' %>
```
[the remark this answers is not in the document]
Comment (davidcam-src) [reply app/views/hyrax/base/_show_actions.html.erb:34]: Gotcha, thanks for explaining this. I'll rollback the change

**Pull request merged**